### PR TITLE
glTF: fix traverse() in group.js and use it for rendering

### PR DESCRIPTION
**Pull request by georgios-uber**

### CODE REVIEW

File: modules/core/src/scenegraph/gltf-instantiator/gltf-instantiator.js [hunk 1]
```diff
@@ -158,6 +169,12 @@ export default class GLTFInstantiator {
       if (material.pbrMetallicRoughness.baseColorTexture) {
         this.configureTexture(material.pbrMetallicRoughness.baseColorTexture, model);
       }
+
+      if (material.pbrMetallicRoughness.baseColorFactor) {
+        model.setUniforms({baseColorFactor: material.pbrMetallicRoughness.baseColorFactor});
```
Comment (ibgreen) [inline modules/core/src/scenegraph/gltf-instantiator/gltf-instantiator.js]: Nit: Calculate the value in a ternary and then set it on a separate line?

File: modules/core/src/scenegraph/group.js [hunk 2]
```diff
@@ -38,31 +38,18 @@ export default class Group extends Node {
...
-  // If visitor returns a truthy value, traversal will be aborted and that value
-  // will be returned from `traverse`. Otherwise `traverse` will return null.
-  traverse(visitor, {modelMatrix: modelMatrixIn = new Matrix4()} = {}) {
+  traverse(visitor, {worldMatrix = new Matrix4()} = {}) {
+    const modelMatrix = new Matrix4(worldMatrix).multiplyRight(this.matrix);
+
     for (const child of this.children) {
-      const {matrix} = child;
-      const modelMatrix = new Matrix4(modelMatrixIn).multiplyRight(matrix);
-      let result;
```
Comment (ibgreen) [inline modules/core/src/scenegraph/group.js]: Any chance we can restore the return logic now that we fixed the traversal?

We will need a couple of test cases for the scenegraph stuff soon.

File: modules/core/src/scenegraph/gltf-instantiator/gltf-instantiator.js [hunk 3]
```diff
@@ -99,12 +101,21 @@ export default class GLTFInstantiator {
...
-        scale: gltfNode.scale
-      });
+      node.matrix.identity();
+
+      if (gltfNode.translation) {
+        node.matrix.translate(gltfNode.translation);
+      }
+
+      if (gltfNode.rotation) {
+        node.matrix.multiplyRight(new Matrix4().fromQuaternion(gltfNode.rotation));
```
Comment (ibgreen) [inline modules/core/src/scenegraph/gltf-instantiator/gltf-instantiator.js]: Nit; calculate the `rotationMatrix` on a separate line?

File: examples/gltf/app.js [hunk 4]
```diff
@@ -59,7 +59,7 @@ const INFO_HTML = `
 `;
 
 export const animationLoopOptions = {
```
Comment (ibgreen) [inline examples/gltf/app.js]: Can we rename `animationLoopOptions` to `appState`?

File: modules/core/src/scenegraph/group.js (hunk 2 above)
Comment (georgios-uber) [reply modules/core/src/scenegraph/group.js]: ❌

**Pull request merged**